build_tools/fetch_artifacts.py

```diff
@@ -0,0 +1,157 @@
+# but those artifacts may not have all required dependencies.
+
+import argparse
+import subprocess
+import sys
+
+GENERIC_VARIANT = "generic"
+
+
+def log(*args, **kwargs):
+    print(*args, **kwargs)
+    sys.stdout.flush()
+
+
+def s3_exec(variant, package, run_id, build_dir):
+    cmd = [
+        "aws",
+        "s3",
+        "cp",
+        f"s3://therock-artifacts/{run_id}/{package}_{variant}.tar.xz",
+        build_dir,
+        "--no-sign-request",
+    ]
+    log(f"++ Exec [{cmd}]")
+    try:
+        subprocess.run(cmd, check=True)
+    except Exception as ex:
+        log(f"Exception when executing [{cmd}]")
+        log(str(ex))
+
+
+def retrieve_base_artifacts(args, run_id, build_dir):
+    base_artifacts = [
+        "core-runtime_run",
+        "core-runtime_lib",
+        "sysdeps_lib",
+        "base_lib",
+        "amd-llvm_run",
+        "amd-llvm_lib",
+        "core-hip_lib",
+        "core-hip_dev",
+        "rocprofiler-sdk_lib",
+        "host-suite-sparse_lib",
+    ]
+    if args.blas or args.all:
+        base_artifacts.append("host-blas_lib")
+
+    for base_artifact in base_artifacts:
+        s3_exec(GENERIC_VARIANT, base_artifact, run_id, build_dir)
+
+
+def retrieve_enabled_artifacts(args, test_enabled, target, run_id, build_dir):
+    base_artifact_path = []
+    if args.blas or args.all:
+        base_artifact_path.append("blas")
+    if args.fft or args.all:
+        base_artifact_path.append("fft")
+    if args.miopen or args.all:
+        base_artifact_path.append("miopen")
+    if args.prim or args.all:
+        base_artifact_path.append("prim")
+    if args.rand or args.all:
+        base_artifact_path.append("rand")
+    if args.rccl or args.all:
+        base_artifact_path.append("rccl")
+
+    enabled_artifacts = []
+    for base_path in base_artifact_path:
+        enabled_artifacts.append(f"{base_path}_lib")
+        if test_enabled:
+            enabled_artifacts.append(f"{base_path}_test")
+
+    for enabled_artifact in enabled_artifacts:
+        s3_exec(f"{target}", enabled_artifact, run_id, build_dir)
+
+
+def run(args):
+    run_id = args.run_id
+    target = args.target
+    build_dir = args.build_dir
+    test_enabled = args.test
+    retrieve_base_artifacts(args, run_id, build_dir)
+    retrieve_enabled_artifacts(args, test_enabled, target, run_id, build_dir)
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(prog="fetch_artifacts")
+    parser.add_argument(
+        "--run-id", type=str, help="GitHub run ID to retrieve artifacts from"
+    )
+
+    parser.add_argument(
+        "--target", type=str, help="Target variant for specific GPU target"
+    )
+
+    parser.add_argument(
+        "--build-dir",
+        type=str,
+        default="build/artifacts",
+        help="Path to the artifact build directory",
+    )
+
+    parser.add_argument(
+        "--test",
+        help="If flagged, test artifacts will be retrieved",
+        action="store_true",
```

Comment:
Always do booleans like this:

```
    parser.add_argument(
        "--remote",
        default=False,
        action=argparse.BooleanOptionalAction,
        help="Updates submodules from remote vs current",
    )
```

This way, you can default them all to True and it will automatically support `--no-test`.